### ISSUE TITLE
Unable to find the network_config in the private cluster

### ISSUE THREAD

**Issue by manojanama** [terraform-google-modules/terraform-google-kubernetes-engine#1802]
### TL;DR

Unable to find the network_config in the private cluster due to which we are unable to update the stateful ip-pod range.

We have observed the network_config in the cluster.tf for  beta-private cluster, which is used to update the ip-pod range.
dynamic "network_config" {
    for_each = length(lookup(each.value, "pod_range", "")) > 0 ? [each.value] : []
    content {
      pod_range = lookup(network_config.value, "pod_range", null)
    }
  }

### Expected behavior

We should have the network_config in the cluster.tf for private-cluster.

### Observed behavior

When we execute the terraform without the network_config , we are unable to update with the new ip pod range.

### Terraform Configuration

```hcl
module "gke" {
   source = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
   version= "19.0.0"

   
node_pools = [
{
      name               = var.node_pool_name_stateful_v3
      initial_node_count = 1
      image_type         = "cos_containerd"
      machine_type       = var.node_pool_machine_type_stateful_v3
      min_count          = 1
      max_count          = 24
      local_ssd_count    = 0
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      autoscaling        = true
      auto_upgrade       = true
      auto_repair        = true
      auto_upgrade       = true
      service_account    = var.node_pool_service_account_stateful_v3
      preemptible        = false
      enable_secure_boot = true
      max_pods_per_node  = var.max_pods_per_node_stateful_v3
      pod_range          = var.node_pool_pod_range_stateful_v3

    }
   ]

   node_pools_tags = {
    all = []

    stateless = var.stateless_tags
    stateful = var.stateful_tags
  }


   node_pools_taints = {
    all = []

    stateless = [
      {
        key    = "node-pool-type"
        value  = "stateless"
        effect = "NO_EXECUTE"
      },
    ]
  }

}
```

### Terraform Version

```sh
version= "19.0.0"
```

**Comment by shubhamkr619** [comment]
added support in this PR: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1803. please follow up
